src/main/java/by/onliner/listener/TestListener.java

```diff
@@ -0,0 +1,185 @@
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import static com.github.automatedowl.tools.AllureEnvironmentWriter.allureEnvironmentWriter;
+
+/***
+ * TestListener is used to log information on the test run flow and to the reporting system.
+ */
+public class TestListener implements ITestListener {
+
+    protected final Logger logger = LogManager.getLogger(this);
+
+    @Override
+    public void onStart(ITestContext context) {
+        logger.info("The test class run started on " + context.getStartDate());
+        allureEnvironmentWriter(
+                ImmutableMap.<String, String>builder()
+                        .put("Browser", "Chrome")
+                        .put("Browser.Version", "96.0.4664.110 (Official build), (64 bit)")
+                        .build(), System.getProperty("user.dir")
+                        + "/target/allure-results/");
+    }
+
+    @Override
+    public void onFinish(ITestContext context) {
+        logger.info("\n" +
+                "  _____ _____ ____ _____   _____ ___ _   _ ___ ____  _   _ _____ ____  \n" +
+                " |_   _| ____/ ___|_   _| |  ___|_ _| \\ | |_ _/ ___|| | | | ____|  _ \\ \n" +
+                "   | | |  _| \\___ \\ | |   | |_   | ||  \\| || |\\___ \\| |_| |  _| | | | |\n" +
+                "   | | | |___ ___) || |   |  _|  | || |\\  || | ___) |  _  | |___| |_| |\n" +
+                "   |_| |_____|____/ |_|   |_|   |___|_| \\_|___|____/|_| |_|_____|____/ \n" +
+                "on -> " + context.getEndDate());
+        try {
+            FileUtils.deleteDirectory(new File(Recorder.savePath));
+            clearFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get Started, this method is called.
+     */
+    @Override
+    public void onTestStart(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] STARTED");
+        try {
+            Recorder.startRecording(result.getName());
+        } catch (Exception e) {
+            logger.info("Can't start video recording");
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get passed, this method is called.
+     */
+    @Override
+    public void onTestSuccess(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] PASSED");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get failed, this method is called.
+     */
+    @Override
+    public void onTestFailure(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] FAILED");
```

Comment:
same comment as above

Reply:
fixed to error

Reply:
Why to debug?
Sorry for not clear comment. Comment relates to string formatting not to logging level

Reply:
Ok, fixed